client/src/ehr_components/NewPatient.tsx

```diff
@@ -0,0 +1,245 @@
+      errorMessage: "Please select a date",
+      formSubmitted: formSubmitted
+    },
+    {
+      label: "Gender",
+      name: "gender",
+      type: "Radio",
+      options: [
+        {
+          value: "male",
+          label: "Male"
+        },
+        {
+          value: "female",
+          label: "Female"
+        },
+        {
+          value: "intersex",
+          label: "Intersex"
+        },
+        {
+          value: "other",
+          label: "Other"
+        },
+        {
+          value: "undisclosed",
+          label: "Undisclosed"
+        }
+      ],
+      onChange: updateFormData,
+      value: formData.gender,
+      validateValue: (value: string) => {
+        return value.length > 0;
+      },
+      errorMessage: "Please select a gender",
+      formSubmitted: formSubmitted
+    },
+    {
+      label: "Phone Number",
+      name: "phoneNumber",
+      type: "Input",
+      inputType: "text",
+      placeholder: "Phone Number",
+      onChange: updateFormData,
+      validateValue: (value: string) => {
+        let countDigitsRegex = /\d/g;
+        let regexMatch = value.match(countDigitsRegex);
+        return regexMatch !== null ? regexMatch.length === 10 : false;
+      },
+      formatValue: (value: string) => {
+        if (!value) return value;
+
+        const phoneNumber = value.replace(/[^\d]/g, "");
+        const phoneNumberLength = phoneNumber.length;
+
+        if (phoneNumberLength < 4) return phoneNumber;
+        if (phoneNumberLength < 7) {
+          return `(${phoneNumber.slice(0, 3)}) ${phoneNumber.slice(3)}`;
+        }
+        return `(${phoneNumber.slice(0, 3)}) ${phoneNumber.slice(
+          3,
+          6
+        )}-${phoneNumber.slice(6)}`;
+      },
+      value: formData.phoneNumber,
+      errorMessage: "Please enter a 10 digit phone number",
+      formSubmitted: formSubmitted
+    },
+    {
+      label: "Address",
+      name: "address",
+      type: "Input",
+      inputType: "text",
+      placeholder: "Address",
+      onChange: updateFormData,
+      validateValue: (value: string) => {
+        return value.length > 0;
+      },
+      value: formData.address,
+      errorMessage: "Please enter a address",
+      formSubmitted: formSubmitted
+    },
+    {
+      label: "Zip Code",
+      name: "zipCode",
+      type: "Input",
+      inputType: "text",
+      placeholder: "Zip Code",
+      onChange: updateFormData,
+      validateValue: (value: string) => {
+        let countDigitsRegex = /\d/g;
+        let regexMatch = value.match(countDigitsRegex);
+        return regexMatch !== null ? regexMatch.length === 5 : false;
+      },
+      value: formData.zipCode,
+      errorMessage: "Please enter a 5 digit zip code",
+      formSubmitted: formSubmitted
+    },
+    {
+      label: "Job",
+      name: "job",
+      type: "Input",
+      inputType: "text",
+      placeholder: "Job",
+      onChange: updateFormData,
+      validateValue: (value: string) => {

```

Comment:
Job field would probably be optional. let's try to implement that

Reply:
Updated in the latest commit!